src/app/GitUI/LeftPanel/FavoritesTree.cs

```diff
@@ -0,0 +1,168 @@
+﻿using GitCommands;
+using GitCommands.Git;
+using GitCommands.Remotes;
+using GitExtensions.Extensibility;
+using GitExtensions.Extensibility.Git;
+using GitUI.CommandDialogs;
+using GitUI.Properties;
+using GitUI.UserControls.RevisionGrid;
+using Microsoft;
+
+namespace GitUI.LeftPanel
+{
+    internal sealed class FavoritesTree : BaseRefTree
+    {
+        private readonly IAheadBehindDataProvider? _aheadBehindDataProvider;
+        private readonly IRevisionGridInfo _revisionGridInfo;
+        private readonly CachedFavorites _cachedFavorites = new();
+        private readonly IGitUICommandsSource _refUiCommand;
+
+        public FavoritesTree(TreeNode treeNode, IGitUICommandsSource uiCommands, IAheadBehindDataProvider? aheadBehindDataProvider, ICheckRefs refsSource, IRevisionGridInfo revisionGridInfo)
+            : base(treeNode, uiCommands, refsSource, RefsFilter.Remotes | RefsFilter.Heads)
+        {
+            _aheadBehindDataProvider = aheadBehindDataProvider;
+            _revisionGridInfo = revisionGridInfo;
+            _refUiCommand = uiCommands;
+
+            if (refsSource is GitModuleControl { Module.WorkingDirGitDir: not null } gitModuleControl)
+            {
+                _cachedFavorites.Location = gitModuleControl.Module.WorkingDirGitDir;
+            }
+        }
+
+        public void Add(NodeBase node)
+        {
+            if (node is BaseRevisionNode baseRevisionNode && baseRevisionNode.ObjectId != null)
+            {
+                _cachedFavorites.Add(baseRevisionNode.ObjectId.ToString());
```

Comment:
`.ObjectId` seems to be the hash of the HEAD commit of the branch. Thus it is inapt for identifying a branch and causes hard-to-grasp effects.
Deriving the key for a branch should be encapsulated in `CachedFavorites`.